NoLoSay/NoLoApp
# Setting up documentation

## What type of PR is this? (check all applicable)

- [x] Other (please precise) : Setup documentation

## Description

Sets up the documentation with [typedoc](https://typedoc.org/guides/overview/)
Creates a custom theme that takes the colors of NoLoSay
Deploy automatically on github pages on push on main branch

## Related Tickets & Documents

- Related Issue [NOLO-166](https://nolosay.atlassian.net/jira/software/c/projects/NOLO/boards/1?modal=detail&selectedIssue=NOLO-166)

## QA Instructions, Screenshots, Recordings

To test it, you can simply run `yarn run documentation` and open the file `docs/documentation/index.html` in your browser.
Or you can go and see the [current documentation](https://nolosay.github.io/NoLoApp)

## Added/updated tests?

- [x] No, and this is why: Not in the scope of the PR

## [optional] What gif best describes this PR or how it makes you feel?

![](https://github.com/NoLoSay/NoLoApp/assets/25590592/51457394-c235-4dfb-a5c3-21eff2b7ff12)

[NOLO-166]: https://nolosay.atlassian.net/browse/NOLO-166?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
